pkg/virtualKubelet/translation/serviceEnv/serviceEnv.go

```diff
@@ -0,0 +1,120 @@
+package serviceEnv
+
+import (
+	apimgmgt "github.com/liqotech/liqo/pkg/virtualKubelet/apiReflection"
+	"github.com/liqotech/liqo/pkg/virtualKubelet/storage"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+	"k8s.io/kubernetes/pkg/kubelet/envvars"
+)
+
+var masterServices = sets.NewString("kubernetes")
+
+func TranslateServiceEnvVariables(pod *v1.Pod, localNS string, nattedNS string, cacheManager storage.CacheManagerReader) (*v1.Pod, error) {
+	resPod := pod.DeepCopy()
+
+	enableServiceLinks := v1.DefaultEnableServiceLinks
+	if resPod.Spec.EnableServiceLinks != nil {
+		enableServiceLinks = *resPod.Spec.EnableServiceLinks
+	}
+
+	envs, err := getServiceEnvVarMap(localNS, enableServiceLinks, nattedNS, cacheManager)
+	if err != nil {
+		return nil, err
+	}
+	for i, container := range resPod.Spec.Containers {
+		resPod.Spec.Containers[i] = *setInContainer(envs, container)
+	}
+	for i, container := range resPod.Spec.InitContainers {
+		resPod.Spec.InitContainers[i] = *setInContainer(envs, container)
+	}
+	return resPod, nil
+}
+
+func setInContainer(envs map[string]string, container v1.Container) *v1.Container {
+	for k, v := range envs {
+		found := false
+		for _, env := range container.Env {
+			if env.Name == k {
+				found = true
+				break
+			}
+		}
+		if !found {
+			container.Env = append(container.Env, v1.EnvVar{
+				Name:  k,
+				Value: v,
+			})
+		}
+	}
+	return container.DeepCopy()
+}
+
+func getServiceEnvVarMap(ns string, enableServiceLinks bool, remoteNs string, cacheManager storage.CacheManagerReader) (map[string]string, error) {
+	var (
+		serviceMap = make(map[string]*v1.Service)
+		m          = make(map[string]string)
+	)
+
+	// search for services in the same namespaces of the pod
+	services, err := cacheManager.ListHomeNamespacedObject(apimgmgt.Services, ns)
+	if err != nil {
+		return nil, err
+	}
+
+	// project the services in namespace ns onto the master services
+	for i := range services {
+		// We always want to add environment variables for master kubernetes service
+		// from the default namespace, even if enableServiceLinks is false.
+		// We also add environment variables for other services in the same
+		// namespace, if enableServiceLinks is true.
+
+		tmp := services[i]
+		service, ok := tmp.(*v1.Service)
+		if !ok {
+			continue
```

Comment:
Can you add a comment with high verbosity (>=4)? For having visibility on this fault (very unlikely to me)